lib/tasks/storage.rake

```diff
@@ -0,0 +1,27 @@
+namespace :storage do
```

Comment:
Not sure this is working as intended:

```
$ bin/rails storage:migrate
[DEPRECATION] AlaveteliConfiguration.raw_emails_location will be removed in 0.42. You should have `ActiveStorage` configured but you still haven't migrated all your old files. Please un `rails storage:migrate` to migrate
Migrated raw email (1/34)[DEPRECATION] AlaveteliConfiguration.raw_emails_location will be removed in 0.42. You should have `ActiveStorage` configured but you still haven't migrated all your old files. Please un `rails storage:migrate` to migrate
Migrated raw email (2/34)[DEPRECATION] AlaveteliConfiguration.raw_emails_location will be removed in 0.42. You should have `ActiveStorage` configured but you still haven't migrated all your old files. Please un `rails storage:migrate` to migrate
Migrated raw email (3/34)[DEPRECATION] AlaveteliConfiguration.raw_emails_location will be removed in 0.42. You should have `ActiveStorage` configured but you still haven't migrated all your old files. Please un `rails storage:migrate` to migrate
Migrated raw email (4/34)[DEPRECATION] AlaveteliConfiguration.raw_emails_location will be removed in 0.42. You should have `ActiveStorage` configured but you still haven't migrated all your old files. Please un `rails storage:migrate` to migrate
Migrated raw email (5/34)[DEPRECATION] AlaveteliConfiguration.raw_emails_location will be removed in 0.42. You should have `ActiveStorage` configured but you still haven't migrated all your old files. Please un `rails storage:migrate` to migrate
Migrated raw email (6/34)[DEPRECATION] AlaveteliConfiguration.raw_emails_location will be removed in 0.42. You should have `ActiveStorage` configured but you still haven't migrated all your old files. Please un `rails storage:migrate` to migrate
Migrated raw email (7/34)[DEPRECATION] AlaveteliConfiguration.raw_emails_location will be removed in 0.42. You should have `ActiveStorage` configured but you still haven't migrated all your old files. Please un `rails storage:migrate` to migrate
Migrated raw email (8/34)[DEPRECATION] AlaveteliConfiguration.raw_emails_location will be removed in 0.42. You should have `ActiveStorage` configured but you still haven't migrated all your old files. Please un `rails storage:migrate` to migrate
Migrated raw email (9/34)[DEPRECATION] AlaveteliConfiguration.raw_emails_location will be removed in 0.42. You should have `ActiveStorage` configured but you still haven't migrated all your old files. Please un `rails storage:migrate` to migrate
Migrated raw email (10/34)[DEPRECATION] AlaveteliConfiguration.raw_emails_location will be removed in 0.42. You should have `ActiveStorage` configured but you still haven't migrated all your old files. Please un `rails storage:migrate` to migrate
Migrated raw email (11/34)[DEPRECATION] AlaveteliConfiguration.raw_emails_location will be removed in 0.42. You should have `ActiveStorage` configured but you still haven't migrated all your old files. Please un `rails storage:migrate` to migrate
Migrated raw email (12/34)[DEPRECATION] AlaveteliConfiguration.raw_emails_location will be removed in 0.42. You should have `ActiveStorage` configured but you still haven't migrated all your old files. Please un `rails storage:migrate` to migrate

# Hung here, so aborted the process…

^Crails aborted!
Interrupt:
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/share_lock.rb:220:in `wait_for'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/share_lock.rb:190:in `block in yield_shares'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/share_lock.rb:189:in `yield_shares'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/dependencies/interlock.rb:47:in `permit_concurrent_loads'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:91:in `block in execute'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract_adapter.rb:728:in `block (2 levels) in log'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract_adapter.rb:727:in `block in log'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract_adapter.rb:718:in `log'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:90:in `execute'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:150:in `begin_db_transaction'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract/transaction.rb:179:in `materialize!'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract/transaction.rb:246:in `block (2 levels) in materialize_transactions'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract/transaction.rb:246:in `each'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract/transaction.rb:246:in `block in materialize_transactions'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract/transaction.rb:243:in `materialize_transactions'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract/database_statements.rb:288:in `materialize_transactions'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:61:in `query'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract/database_statements.rb:102:in `query_values'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract/schema_statements.rb:37:in `data_sources'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/schema_cache.rb:137:in `prepare_data_sources'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/schema_cache.rb:54:in `data_source_exists?'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/type_caster/connection.rb:20:in `type_for_attribute'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/table_metadata.rb:33:in `type'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/predicate_builder.rb:51:in `build'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/predicate_builder.rb:109:in `block in expand_from_hash'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/predicate_builder.rb:68:in `each'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/predicate_builder.rb:68:in `flat_map'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/predicate_builder.rb:68:in `expand_from_hash'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/predicate_builder.rb:70:in `block in expand_from_hash'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/predicate_builder.rb:68:in `each'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/predicate_builder.rb:68:in `flat_map'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/predicate_builder.rb:68:in `expand_from_hash'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/predicate_builder.rb:21:in `build_from_hash'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/where_clause_factory.rb:19:in `build'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/query_methods.rb:656:in `where!'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/associations/association_scope.rb:162:in `apply_scope'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/associations/association_scope.rb:65:in `last_chain_scope'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/associations/association_scope.rb:122:in `add_constraints'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/associations/association_scope.rb:29:in `scope'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/associations/association_scope.rb:7:in `scope'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/associations/association.rb:224:in `association_scope'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/associations/association.rb:101:in `scope'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/associations/association.rb:204:in `find_target'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/associations/singular_association.rb:39:in `find_target'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/associations/association.rb:164:in `load_target'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/associations/association.rb:65:in `reload'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/associations/singular_association.rb:9:in `reader'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/associations/builder/association.rb:100:in `file_blob'
/home/vagrant/bundle/ruby/2.7.0/gems/activestorage-6.0.4.1/lib/active_storage/attached/changes/create_one.rb:42:in `public_send'
/home/vagrant/bundle/ruby/2.7.0/gems/activestorage-6.0.4.1/lib/active_storage/attached/changes/create_one.rb:42:in `find_attachment'
/home/vagrant/bundle/ruby/2.7.0/gems/activestorage-6.0.4.1/lib/active_storage/attached/changes/create_one.rb:38:in `find_or_build_attachment'
/home/vagrant/bundle/ruby/2.7.0/gems/activestorage-6.0.4.1/lib/active_storage/attached/changes/create_one.rb:15:in `attachment'
/home/vagrant/bundle/ruby/2.7.0/gems/activestorage-6.0.4.1/lib/active_storage/attached/changes/create_one.rb:32:in `save'
/home/vagrant/bundle/ruby/2.7.0/gems/activestorage-6.0.4.1/lib/active_storage/attached/model.rb:55:in `block in has_one_attached'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/callbacks.rb:428:in `instance_exec'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/callbacks.rb:428:in `block in make_lambda'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/callbacks.rb:238:in `block in halting_and_conditional'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/callbacks.rb:517:in `block in invoke_after'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/callbacks.rb:517:in `each'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/callbacks.rb:517:in `invoke_after'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/callbacks.rb:136:in `run_callbacks'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/callbacks.rb:825:in `_run_save_callbacks'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/callbacks.rb:327:in `create_or_update'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/timestamp.rb:128:in `create_or_update'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/persistence.rb:470:in `save'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/validations.rb:47:in `save'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/transactions.rb:314:in `block in save'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/transactions.rb:375:in `block in with_transaction_returning_status'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract/database_statements.rb:278:in `transaction'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/transactions.rb:212:in `transaction'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/transactions.rb:366:in `with_transaction_returning_status'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/transactions.rb:314:in `save'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/suppressor.rb:44:in `save'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/persistence.rb:621:in `block in update'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/transactions.rb:375:in `block in with_transaction_returning_status'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract/database_statements.rb:280:in `block in transaction'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract/transaction.rb:280:in `block in within_new_transaction'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
/home/vagrant/bundle/ruby/2.7.0/gems/activesupport-6.0.4.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract/transaction.rb:278:in `within_new_transaction'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/connection_adapters/abstract/database_statements.rb:280:in `transaction'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/transactions.rb:212:in `transaction'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/transactions.rb:366:in `with_transaction_returning_status'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/persistence.rb:619:in `update'
/home/vagrant/bundle/ruby/2.7.0/gems/activestorage-6.0.4.1/lib/active_storage/attached/one.rb:32:in `attach'
/home/vagrant/alaveteli/app/models/raw_email.rb:119:in `data='
/home/vagrant/alaveteli/lib/tasks/storage.rake:19:in `block (3 levels) in <top (required)>'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/batches.rb:70:in `block (2 levels) in find_each'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/batches.rb:70:in `each'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/batches.rb:70:in `block in find_each'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/batches.rb:136:in `block in find_in_batches'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/batches.rb:238:in `block in in_batches'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/batches.rb:222:in `loop'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/batches.rb:222:in `in_batches'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/batches.rb:135:in `find_in_batches'
/home/vagrant/bundle/ruby/2.7.0/gems/activerecord-6.0.4.1/lib/active_record/relation/batches.rb:69:in `find_each'
/home/vagrant/alaveteli/lib/tasks/storage.rake:18:in `with_index'
/home/vagrant/alaveteli/lib/tasks/storage.rake:18:in `block (2 levels) in <top (required)>'
/home/vagrant/bundle/ruby/2.7.0/gems/railties-6.0.4.1/lib/rails/commands/rake/rake_command.rb:23:in `block in perform'
/home/vagrant/bundle/ruby/2.7.0/gems/railties-6.0.4.1/lib/rails/commands/rake/rake_command.rb:20:in `perform'
/home/vagrant/bundle/ruby/2.7.0/gems/railties-6.0.4.1/lib/rails/command.rb:48:in `invoke'
/home/vagrant/bundle/ruby/2.7.0/gems/railties-6.0.4.1/lib/rails/commands.rb:18:in `<top (required)>'
bin/rails:4:in `require'
bin/rails:4:in `<main>'
Tasks: TOP => storage:migrate
(See full trace by running task with --trace)
```

Second run ran through to completion, but with the deprecation output.

Reply:
Of course, will silence the depreciation warning for this task.

Reply:
Fixed in 7a63b968c